Src/VimMac/CaretUtil.cs

```diff
@@ -46,9 +47,39 @@ private void SetCaret(IVimBuffer vimBuffer)
 
         void IVimBufferCreationListener.VimBufferCreated(IVimBuffer vimBuffer)
         {
-            SetCaret(vimBuffer);
-            vimBuffer.SwitchedMode += (_,__) => SetCaret(vimBuffer);
-            _inlineRenameListenerFactory.RenameUtil.IsRenameActiveChanged += (_, __) => SetCaret(vimBuffer);
+            _vimBuffer = vimBuffer;

```

Comment:
If multiple IVimBuffer are created they will overwrite this variable. Believe that causes the later event handlers to potentially operate on the wrong instance. Basically is this okay if two IVimBuffer are created and then the first one is closed. Won’t that cause the event handlers to make changes to second one?

Reply:
Ah, yes. Good spot!